src/App.js

```diff
@@ -1,13 +1,126 @@
 import React from 'react';
+import TodoForm from './components/TodoComponents/TodoForm';
+import TodoList from './components/TodoComponents/TodoList';
+import './App.css';
+import TodoHeader from './components/TodoComponents/TodoHeader';
+import TodoMessage from './components/TodoComponents/TodoMessage';
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      input: '',
+      todos: [],
+      todoStyle: 'Todo',
+      toDoCompleteStyle: { textDecoration: 'line-through' },
+      showMessage: { display: 'block' }
+    };
+  }
+
+  handleNewToDo = e => {
+    e.preventDefault();
+    // !this.state.input is equivalent to this.state.input === '' (falsey)
+    if (!this.state.input) {
+      this.setState({ invalidInput: true });
+    } else {
+      const task = {
+        // slice creates a copy of this.state.input
+        task: this.state.input.slice(),
```

Comment:
don't need to create a copy of this.state.input, it is a string, and they are immutable, so they are always copied 

https://codeburst.io/explaining-value-vs-reference-in-javascript-647a975e12a0